contracts/pallet-ibc/src/ics20/mod.rs

```diff
@@ -561,11 +563,370 @@ pub fn full_ibc_denom(packet: &Packet, mut token: PrefixedCoin) -> String {
+			Ics20Error::implementation_specific("Failed to get memo".to_string())
+		})?;
+
+		let memo_forward = match memo_forward {
+			MemoType::IBC(memo_forward) => memo_forward,
+			MemoType::XCM(memo_forward) => {
+				let s = memo_forward.receiver.strip_prefix("0x").ok_or_else(|| {
+					Self::emit_memo_execution_failed_event(
+						receiver.clone(),
+						packet_data.memo.clone(),
+						11,
+					);
+					Ics20Error::implementation_specific("failed strip_prefix.".to_string())
+				})?;
+
+				let decoded_accout = hex::decode(s).map_err(|_| {
+					Self::emit_memo_execution_failed_event(
+						receiver.clone(),
+						packet_data.memo.clone(),
+						12,
+					);
+					Ics20Error::implementation_specific("hex::decode".to_string())
+				})?;
+
+				let account_to = T::AccountId::decode(&mut &*decoded_accout).map_err(|_| {
+					Self::emit_memo_execution_failed_event(
+						receiver.clone(),
+						packet_data.memo.clone(),
+						13,
+					);
+					Ics20Error::implementation_specific("T::AccountId::decode".to_string())
+				})?;
+
+				T::SubstrateMultihopXcmHandler::transfer_xcm(
+					receiver.clone(),
+					account_to.clone(),
+					memo_forward.para_id,
+					amount,
+					asset_id.into(),
+				)
+				.ok_or_else(|| {
+					crate::Pallet::<T>::deposit_event(Event::<T>::ExecuteMemoXcmFailed {
+						from: receiver.clone(),
+						to: account_to.clone(),
+						para_id: memo_forward.para_id,
+						amount,
+						asset_id: asset_id.into(),
+					});
+					Ics20Error::implementation_specific(
+						"Faield to execute SubstrateMultihopXcmHandler::transfer_xcm.".to_string(),
+					)
+				})?;
+
+				crate::Pallet::<T>::deposit_event(Event::<T>::ExecuteMemoXcmSuccess {
+					from: receiver.clone(),
+					to: account_to.clone(),
+					para_id: memo_forward.para_id,
+					amount,
+					asset_id: asset_id.into(),
+				});
+
+				return Ok(())
+			},
+		};
+
+		// let raw_bytes = memo_forward.receiver.as_bytes().to_vec();
+		let raw_bytes = memo_forward.receiver.into_bytes();
+
+		//convert string into T::AccountId
+
+		let transfer_ibc_account_to =
+			crate::MultiAddress::<<T as frame_system::Config>::AccountId>::Raw(raw_bytes.clone());
+
+		let origin = RawOrigin::Signed(receiver.clone());
+		let channel_id = memo_forward
+			.channel
+			.split('-')
+			.last()
+			.ok_or_else(|| {
+				Self::emit_memo_execution_failed_event(
+					receiver.clone(),
+					packet_data.memo.clone(),
+					3,
+				);
+				Ics20Error::implementation_specific(format!(
+					"Failed to extract channel number from channel ID: {:?}",
+					memo_forward.channel
+				))
+			})?
+			.parse()
+			.map_err(|_| {
+				Self::emit_memo_execution_failed_event(
+					receiver.clone(),
+					packet_data.memo.clone(),
+					4,
+				);
+				Ics20Error::implementation_specific("Failed to parse channel ID".to_string())
+			})?;
+
+		//timestamp that current timestamp + 30 minutes
+		let _timestamp = <pallet_timestamp::Pallet<T> as frame_support::traits::UnixTime>::now()
```

Comment:
The variable is not used anywhere, remove?

Reply:
Done.